sphinx/content/ch1.md

```diff
@@ -0,0 +1,1249 @@
+---
+array2d = np.array(expression_data)
+print(array2d)
+print(array2d.shape)
+print(type(array2d))
+```
+
+Now you can see that the `shape` attribute generalizes `len` to account for the size of multiple dimensions of an array of data.
+
+```{figure} ../../figures/NumPy_ndarrays_v2.png
+---
+name: fig:ndary
+---
+Visualizing NumPy's ndarrays in one, two and three dimensions
+```
+
+Arrays have other attributes, such as `ndim`, the number of dimensions:
+
+```python
+print(array2d.ndim)
+```
+
+You'll become familiar with all of these as you start to use NumPy more for your own data analysis.
+
+NumPy arrays can represent data that has even more dimensions ({numref}`fig:ndary`), such as magnetic resonance imaging (MRI) data, which includes measurements within a 3D volume.
+If we store MRI values over time, we might need a 4D NumPy array.
+
+For now, we'll stick to 2D data.
+Later chapters will introduce higher-dimensional data and will teach you to write code that works for data of any number of dimensions.
+
+### Why Use ndarrays Instead of Python Lists?
+
+Arrays are fast because they enable *vectorized* operations, written in the low-level language C, that act on the whole array.
+Say you have a list and you want to multiply every element in the list by five.
+A standard Python approach would be to write a loop that iterates over the
+elements of the list and multiply each one by five.
+However, if your data were instead represented as an array,
+you can multiply every element in the array by five in a single bound.
+Behind the scenes, the highly-optimized NumPy library is doing the iteration as fast as possible.
+
+```python
+import numpy as np
+
+# Create an ndarray of integers in the range
+# 0 up to (but not including) 1,000,000
+array = np.arange(1e6)
+
+# Convert it to a list
+list_array = array.tolist()
+```
+
+Let's compare how long it takes to multiply all the values in the array by five,
+using the IPython `timeit` magic function. First, when the data is in a list:
+
+```ipython
+%timeit -n10 y = [val * 5 for val in list_array]
+```
+
+Now, using NumPy's built-in *vectorized* operations:
+
+```ipython
+%timeit -n10 x = array * 5
+```
+
+Over 50 times faster, and more concise, too!
+
+Arrays are also size efficient.
+In Python, each element in a list is an object and is given a healthy memory allocation (or is that unhealthy?).
+In contrast, in arrays, each element takes up just the necessary amount of memory.
+For example, an array of 64-bit integers takes up exactly 64-bits per element, plus some very small overhead for array metadata, such as the `shape` attribute we discussed above.
+This is generally much less than would be given to objects in a Python list.
+```{note}
+If you're interested in digging into how Python memory allocation works,
+check out Jake VanderPlas's blog post:
+["Why Python Is Slow: Looking Under the Hood"](https://jakevdp.github.io/blog/2014/05/09/why-python-is-slow/).
```

Comment:
Why the quotes here?

Reply:
No explicit reason - just a little piece that got lost in translation. Thanks for pointing it out!